WooCommerce/Classes/Authentication/Store Creation/Progress/StoreCreationProgressViewModel.swift

```diff
@@ -0,0 +1,158 @@
+import UIKit
+
+final class StoreCreationProgressViewModel: ObservableObject {
+    /// Conforms to `Double` to use a value for
+    /// `ProgressView` in `StoreCreationProgressView`
+    ///
+    enum Progress: Double, CaseIterable {
+        case creatingStore = 0.0
+        case buildingFoundations = 1.0
+        case organizingStockRoom = 2.0
+        case applyingFinishingTouches = 3.0
+        case finished = 4.0
+    }
+
+    @Published private var progress: Progress = .creatingStore
+
+    let totalProgressAmount = StoreCreationProgressViewModel.Progress.finished.rawValue
+
+    @Published private(set) var progressValue: Double = StoreCreationProgressViewModel.Progress.creatingStore.rawValue
+
+    var title: String {
+        progress.title
+    }
+
+    var subtitle: AttributedString {
+        var prefix = AttributedString(Localization.funWooFact)
+        prefix.font = .headline
+        prefix.foregroundColor = .init(.text)
+
+        var attributedText = AttributedString(progress.subtitle)
+        attributedText.font = .body
+        attributedText.foregroundColor = .init(.text)
+
+        return prefix + " " + attributedText
+    }
+
+    private var incrementProgressValueTimer: Timer?
+    private let expectNextIncrementToOccurAfter: Double
+    private let progressViewAnimationTimerInterval: TimeInterval
+
+    /// - Parameters:
+    ///   - expectNextIncrementToOccurAfter:
+    ///     Approx interval at which progress will be incremented to next case.
+    ///     This value is used to animate the progress view until next increment happens.
+    ///
+    ///   - progressViewAnimationTimerInterval: Animation timer interval DI for unit test purposes.
+    ///
+    init(expectNextIncrementToOccurAfter: TimeInterval,
```

Comment:
nit: how about something like `estimatedTimeForEachProgress`/`estimatedProgressTime`/`estimatedTimePerProgress` at least for the internal property?

Reply:
Renamed in 61f28b8cf57d92a7ce631e8c4a7cdb8a2061f70f